### PR TITLE
Improve Shopware landing page

### PR DESCRIPTION
**Pull request by JaroslavSku**

### CODE REVIEW

File: modules/cms-shopware/components/SfxShopwareCmsPage.vue [hunk 1]
```diff
@@ -42,7 +42,7 @@ export default defineComponent({
 
     for (const section of this.data.sections) {
       for (const block of section.blocks) {
-        renderedBlock.push(this.renderBlock(block))
+        renderedBlock.push(this.renderBlock({ ...block, ...section }))
```
Comment (adamjedlicka) [inline modules/cms-shopware/components/SfxShopwareCmsPage.vue:45]: Wouldnt it be better to change renderBlock to accept object of block and section?

**Review by adamjedlicka — CHANGES_REQUESTED** [review]

File: modules/cms-shopware/components/SfxShopwareCmsPage.vue [hunk 2]
```diff
@@ -11,19 +11,20 @@ export default defineComponent({
   },
 
   methods: {
-    renderBlock(block: any) {
-      const component = this.component(block.type)
+    renderBlock(group: any) {
```
Comment (adamjedlicka) [inline modules/cms-shopware/components/SfxShopwareCmsPage.vue:14]: Use object spread here `renderBlock({ block, section })`

File: modules/cms-shopware/components/SfxShopwareCmsPage.vue [hunk 3]
```diff
@@ -11,19 +11,20 @@ export default defineComponent({
...
 
-      for (const slot of block.slots ?? []) {
+      for (const slot of properties.slots ?? []) {
         const slotName = slot.slot === 'content' ? 'default' : slot.slot || 'default'
 
         slots[slotName] = () => this.renderBlock(slot)
       }
 
-      return h(component, { data: block }, slots)
+      return h(component, { data: properties }, slots)
```
Comment (adamjedlicka) [inline modules/cms-shopware/components/SfxShopwareCmsPage.vue:27]: Why not pass section as secondary prop? `{ data: block, section }`

**Review by adamjedlicka — CHANGES_REQUESTED** [review]

File: modules/cms-shopware/components/SfxShopwareCmsPage.vue [hunk 4]
```diff
@@ -11,19 +11,19 @@ export default defineComponent({
   },
 
   methods: {
-    renderBlock(block: any) {
-      const component = this.component(block.type)
+    renderBlock(group: any) {
```
Comment (adamjedlicka) [inline modules/cms-shopware/components/SfxShopwareCmsPage.vue:14]: Change to:

```
renderBlock({ block, section })
```

File: modules/cms-shopware/components/SfxShopwareCmsPage.vue [hunk 5]
```diff
@@ -11,19 +11,19 @@ export default defineComponent({
...
 
-      for (const slot of block.slots ?? []) {
+      for (const slot of group.slots ?? []) {
         const slotName = slot.slot === 'content' ? 'default' : slot.slot || 'default'
 
         slots[slotName] = () => this.renderBlock(slot)
       }
 
-      return h(component, { data: block }, slots)
+      return h(component, { data: group, section: group.section }, slots)
```
Comment (adamjedlicka) [inline modules/cms-shopware/components/SfxShopwareCmsPage.vue:26]: Change to:

```
return h(component, { data: block, section }, slots)
```

File: modules/cms-shopware/components/SfxShopwareCmsPage.vue [hunk 6]
```diff
@@ -42,7 +42,7 @@ export default defineComponent({
 
     for (const section of this.data.sections) {
       for (const block of section.blocks) {
-        renderedBlock.push(this.renderBlock(block))
+        renderedBlock.push(this.renderBlock({ ...block, section }))
```
Comment (adamjedlicka) [inline modules/cms-shopware/components/SfxShopwareCmsPage.vue:45]: Change to:

```
renderedBlock.push(this.renderBlock({ block, section }))
```

**Review by adamjedlicka — CHANGES_REQUESTED** [review]

**Comment by JaroslavSku** [comment]
I created a new one with feature-sfx-108

**Pull request closed without merging**